### DISCUSSION TITLE
Use other spinner in UMD app

### DISCUSSION THREAD

**Discussion by bitmandk** [quasarframework/quasar#10634] [General - Components / Directives / etc] [answered]
I need to show a spinner, using the loading plugin in my UMD app.

The below code works - showing the default spinner:
app.use(Quasar, {
    config: {
        loading: {
            message: 'Please wait!'
        }
    }
})

But I want to use another spinner an I want to set it via the "spinner" property, but spinner won't show when setting the spinner property.

Have tried the following:

- spinner: 'QSpinnerAudio'
- spinner: 'q-spinner-audio'
- spinner: QSpinnerAudio
- spinner: 'spinner-audio'

What is the correct naming for spinners in UMD?

**Comment by ontwikkelfabriek** [comment]
@bitmandk 

> But I want to use another spinner an I want to set it via the "spinner" property,

spinner property of what exactly?

**Reply by bitmandk** [reply]
It's the loading plugin - it shold be possible to use another spinner icon/image

**Reply by ontwikkelfabriek** [reply]
@bitmandk 

1. Could you post the relevant JS `errors` from the debug console?
2. Could you post the entire block of `config` when you attempt to set a custom spinner component for the `loading` plugin?

```javascript
config: {
loading: {
....
```

btw if you create a codepen.io to demonstrate your problem, you will get better support.

**Reply by bitmandk** [reply]
This CodePen shows the code using the default spinner without setting the "spinner" property:

[Codepen](https://codepen.io/bobittjek/pen/qBjmWYZ)

**Reply by ontwikkelfabriek** [reply]
@bitmandk 

Here you go:
https://codepen.io/ontwikkelfabriek/pen/KKqmPjL

The spinner property is of type `Component` ( not a string) , see the API:
https://quasar.dev/quasar-plugins/loading

**QuasarConfOptions**
```
spinner : Component
Description
One of the QSpinners
```

**Reply by bitmandk** [reply]
Yes I realised that it was a component and not a string - but did not know I had to pre-fix the component name with "Quasar" ;-)

Thank you.

**Reply by ontwikkelfabriek** [reply]
@bitmandk 

> but did not know I had to pre-fix the component name with "Quasar" 

Neither did I.  But when I did `console.log(Quasar)` you'll see that all components/plugins are inside the Quasar object.

**Reply by bitmandk** [reply]
Funny that I didn't think of that - in order to get the loading working I checked the console and noticed that I had to prefix the loading.show() with Quasar: Quasar.Loading.show() - In the CLI version it's $q.loading.show()

**Comment by ontwikkelfabriek** [comment]
@bitmandk 

Can I ask you what your experience/opinion is, when using Quasar in UMD mode ( compared to CLI )? Do you recommend it?

**Reply by bitmandk** [reply]
I have not used the CLI version - I prefer using it stand-alone - great to work with - but many examples are for the CLI version, so sometimes (as in this case) I have problems getting the CLI examples to work in UMD.

**Comment by pdanpdan** [comment]
It's a different use case, as explained in the docs.

**Comment by rstoenescu** [comment, marked as answer]
spinner: Quasar.QSpinnerAudio

**Reply by NicoP-S** [reply]
whats the correct one for quasar cli?

I tried the following but it's not working 
```
const { QSpinnerAudio } = require("quasar");
.....

loading: {
    spinner: QSpinnerAudio
}
````

**Reply by rstoenescu** [reply]
@NicoP-S `import { QSpinner } from 'quasar'`

**Reply by NicoP-S** [reply]
Thanks for the quick answer but this is  not working in the quasar.conf.js
`SyntaxError: Cannot use import statement outside a module`

**Reply by rstoenescu** [reply]
Why do you need it in quasar.conf.js. Doesn't makes any sense.

**Reply by NicoP-S** [reply]
It's from the doc's for setting up defaults. Or did I misunderstand this section?
https://quasar.dev/quasar-plugins/loading#setting-up-defaults

**Reply by MilosPaunovic** [reply]
In `quasar.conf.js` you need to find `framework` attribute, and inside set your defaults:

```js
framework: {
   config: { 
      loading: { 
         delay: 100,
         // ...more options
      }
   }
}
```

**Reply by NicoP-S** [reply]
That is exacly what i did. Other attributes like size and delay are working fine.
The `spinner` attribute needs a component. And that's my problem.

following two things does not work on top of quasar.conf.js
```
import { QSpinner } from 'quasar'
const { QSpinnerAudio } = require("quasar");
```

**Reply by rstoenescu** [reply]
What you need to understand is the context in which quasar.conf.js is executed. It's not your app's code, it's the Quasar CLI running your app config file in Nodejs.

Wherever else in quasar.conf.js you need to specify a component, you just do it in String form:

```
framework: {
   config: { 
      loading: { 
         spinner: 'QSpinnerAudio'
      }
   }
}
```

**Reply by NicoP-S** [reply]
Sorry for replying again. I tried your approuch @rstoenescu .
There is no error but also there is no spinner. I am doing something wrong :(
Here my framework snipped from quasar.conf.js
```
framework: {
  config: {
    loading: {
      spinner: 'QSpinnerAudio'
    }
  },
  lang: 'de', // Quasar language pack
  // Quasar plugins
  plugins: ['Notify', 'Cookies', 'Dialog', 'Loading']
},
```
 » Dev mode.......... spa
 » Pkg quasar........ v2.1.4
 » Pkg @quasar/app... v3.1.

Results in:
![image](https://user-images.githubusercontent.com/8988747/137588410-a62e4cdd-ba99-440f-9070-3f8bbdc297de.png)

**Reply by rstoenescu** [reply]
Add framework > components: [ 'QSpinnerAudio' ] too.

**Reply by NicoP-S** [reply]
still the same result as before... does this work on your side?

**Reply by rstoenescu** [reply]
@NicoP-S

Found a bug on why it's not working as it should.
Pushed a fix and it will be available in "@quasar/app" v3.1.5.
You'll only have to configure quasar.conf.js > framework > config > loading > spinner: 'QSpinnerAudio'
